drivers/pds/targetcluster/targetcluster.go

```diff
@@ -69,25 +69,29 @@ type TargetCluster struct {
 
 func (tc *TargetCluster) GetDeploymentTargetID(clusterID, tenantID string) (string, error) {
 	log.InfoD("Get the Target cluster details")
-	targetClusters, err := components.DeploymentTarget.ListDeploymentTargetsBelongsToTenant(tenantID)
-	var targetClusterStatus string
-	if err != nil {
-		return "", fmt.Errorf("error while listing deployments: %v", err)
-	}
-	if targetClusters == nil {
-		return "", fmt.Errorf("target cluster passed is not available to the account/tenant %v", err)
-	}
-	for i := 0; i < len(targetClusters); i++ {
-		if targetClusters[i].GetClusterId() == clusterID {
-			deploymentTargetID = targetClusters[i].GetId()
-			log.Infof("deploymentTargetID %v", deploymentTargetID)
-			log.InfoD("Cluster ID: %v, Name: %v,Status: %v", targetClusters[i].GetClusterId(), targetClusters[i].GetName(), targetClusters[i].GetStatus())
-			targetClusterStatus = targetClusters[i].GetStatus()
+	err = wait.Poll(DefaultRetryInterval, MaxTimeout, func() (bool, error) {
+		targetClusters, err := components.DeploymentTarget.ListDeploymentTargetsBelongsToTenant(tenantID)
+		var targetClusterStatus string
+		if err != nil {
+			return true, fmt.Errorf("error while listing deployments: %v", err)
 		}
-	}
-	if targetClusterStatus != "healthy" {
-		return "", fmt.Errorf("target Cluster is not in healthy state due to error : %v", err)
-	}
+		if targetClusters == nil {
+			return true, fmt.Errorf("target cluster passed is not available to the account/tenant %v", err)
```

Comment:
return false, why are you returning true ? you need to return true once the condition is satisfied.

Reply:
and also error will be nil here, no need to pass err